packages/ember/src/validation/validators/phone.ts

```diff
@@ -0,0 +1,227 @@
+   */
+  allowBlank?: boolean;
+  /**
+   * If `allow`, the number may contain an area code.
+   * If `disallow`, the number must not contain an area code.
+   * If `require`, the number must contain an area code.
+   * @default 'allow'
+   */
+  areaCode?: 'allow' | 'disallow' | 'require';
+  /**
+   * If `allow`, the number may contain a country code.
+   * If `disallow`, the number must not contain a country code.
+   * If `require`, the number must contain a country code.
+   * @default 'allow'
+   */
+  countryCode?: 'allow' | 'disallow' | 'require';
+  /**
+   * If provided, the area code of the number must not match any of the provided area codes.
+   */
+  invalidAreaCodes?: string[];
+  /**
+   * If provided, the country code of the number must not match any of the provided country codes.
+   */
+  invalidCountryCodes?: string[];
+  /**
+   * If provided, the exchange code of the number must not match any of the provided exchange codes.
+   */
+  invalidExchangeCodes?: string[];
+  /**
+   * If provided, the area code of the number must match one of the provided area codes.
+   */
+  validAreaCodes?: string[];
+  /**
+   * If provided, the country code of the number must match one of the provided country codes.
+   */
+  validCountryCodes?: string[];
+  /**
+   * If provided, the exchange code of the number must match one of the provided exchange codes.
+   */
+  validExchangeCodes?: string[];
+} & BaseOptions;
+
+export default class PhoneValidator<
+  Model extends object,
+  Context extends object = Record<string, unknown>,
+> extends BaseValidator<
+  string | null | undefined,
+  Model,
+  Context,
+  PhoneOptions
+> {
+  defaultOptions = {
+    areaCode: 'allow' as const,
+    countryCode: 'allow' as const,
+  };
+
+  validate(
+    value: string | null | undefined,
+    options: PhoneOptions,
+  ): ValidateFnResponse {
+    const { allowBlank } = options;
+
+    const key = 'nrg.validation.phone.invalid';
+
+    if (isEmpty(value) || isNone(value)) {
+      if (allowBlank) {
+        return true;
+      }
+      return {
+        key,
+        value,
+      };
+    }
+
+    let result = this.checkCountryCode(value, options);
+    if (result !== true) {
+      return result;
+    }
+
+    result = this.checkAreaCode(value, options);
+    if (result !== true) {
+      return result;
+    }
+
+    result = this.checkExchangeCode(value, options);
+    if (result !== true) {
+      return result;
+    }
+
+    if (value) {
+      const length = value.length;
+
+      if (length < 7 || length > 13) {
+        return {
+          key,
+          value,
+        };
+      }
+    }
+
+    return result;
+  }
+
+  checkCountryCode(value: string, options: PhoneOptions): ValidateFnResponse {
+    const { countryCode, validCountryCodes, invalidCountryCodes } = options;
+    const cc = getCountryCode(value);
+
+    if (countryCode !== 'allow' || validCountryCodes || invalidCountryCodes) {
```

Comment:
Is this wrapping if statement necessary?
I think the logic works exactly the same if it is removed.